### PR TITLE
Added RelatedVideos Parsing.

### PR DESCRIPTION
**Pull request by MrF1yn**

### CODE REVIEW

File: library/src/main/java/io/sfrei/tracksearch/tracks/deserializer/YouTubeTrackDeserializer.java [hunk 1]
```diff
@@ -29,24 +29,30 @@ protected YouTubeTrackDeserializer(Class<?> vc) {
...
     public YouTubeTrack deserialize(final JsonParser p, final DeserializationContext ctxt) throws IOException {
-
         final JsonElement rootElement = JsonElement.of(ctxt.readTree(p));
-
         // Track
 
         final String ref = rootElement.getAsString("videoId");
-        final String title = rootElement.get("title", "runs").getFirstField().getAsString("text");
+        String title = rootElement.get("title", "runs").getFirstField().getAsString("text");
+        if (title==null)
```
Comment (s-frei) [inline library/src/main/java/io/sfrei/tracksearch/tracks/deserializer/YouTubeTrackDeserializer.java:37]: We should use the `JsonElement` functionality here.

File: library/src/main/java/io/sfrei/tracksearch/tracks/deserializer/YouTubeTrackDeserializer.java [hunk 2]
```diff
@@ -29,24 +29,30 @@ protected YouTubeTrackDeserializer(Class<?> vc) {
...
+        }
 
         final String url = YouTubeClient.HOSTNAME.concat("/watch?v=").concat(ref);
 
         // Metadata
 
-        final JsonElement owner = rootElement.get("ownerText", "runs").getFirstField();
+        JsonElement owner = rootElement.get("ownerText", "runs").getFirstField();
+
+        if (!owner.present()){
```
Comment (s-frei) [inline library/src/main/java/io/sfrei/tracksearch/tracks/deserializer/YouTubeTrackDeserializer.java:52]: Same as above. Maybe extend the `JsonElement`.

File: library/src/main/java/io/sfrei/tracksearch/tracks/deserializer/YouTubeTrackDeserializer.java [hunk 3]
```diff
@@ -29,24 +29,30 @@ protected YouTubeTrackDeserializer(Class<?> vc) {
 
     @Override
     public YouTubeTrack deserialize(final JsonParser p, final DeserializationContext ctxt) throws IOException {
```
Comment (s-frei) [inline library/src/main/java/io/sfrei/tracksearch/tracks/deserializer/YouTubeTrackDeserializer.java:31]: Revert the fromatting changes please.

File: library/src/main/java/io/sfrei/tracksearch/clients/youtube/YouTubeService.java [hunk 4]
```diff
@@ -27,4 +27,13 @@ Call<ResponseWrapper> getForUrlWithParams(@Url String url,
     Call<ResponseWrapper> getSearchForKeywords(@Query("search_query") String search,
                                                @QueryMap Map<String, String> params);
 
+    @GET("/watch")
+    @Headers({
+            TrackSearchConfig.HEADER_LANGUAGE_ENGLISH,
+            TrackSearchConfig.HEADER_YOUTUBE_CLIENT_NAME,
+            TrackSearchConfig.HEADER_YOUTUBE_CLIENT_VERSION
+    })
+    Call<ResponseWrapper> getVideoPage(@Query("v") String search,
```
Comment (s-frei) [inline library/src/main/java/io/sfrei/tracksearch/clients/youtube/YouTubeService.java:36]: Passed `String` should be named `videoID`.

File: library/src/main/java/io/sfrei/tracksearch/clients/soundcloud/SoundCloudClient.java [hunk 5]
```diff
@@ -176,6 +173,11 @@ public TrackList<SoundCloudTrack> getTracksForSearch(@NonNull final String searc
         return trackList;
     }
 
+    @Override
+    public TrackList<YouTubeTrack> getRelatedTracks(@NonNull String videoID) throws TrackSearchException {
```
Comment (s-frei) [inline library/src/main/java/io/sfrei/tracksearch/clients/soundcloud/SoundCloudClient.java:177]: The `SoundCloudClient` shouldn't offert anything related to *YoutTube*

Comment (s-frei) [reply library/src/main/java/io/sfrei/tracksearch/clients/soundcloud/SoundCloudClient.java:177]: *SoundCloud* has also the *related tracks*, this should be added.

File: library/src/main/java/io/sfrei/tracksearch/clients/MultiSearchClient.java [hunk 6]
```diff
@@ -43,6 +43,11 @@ public TrackList<Track> getTracksForSearch(@NonNull final String search) throws
         return getTracksForSearch(search, clients);
     }
 
+    @Override
+    public TrackList<YouTubeTrack> getRelatedTracks(@NonNull String videoID) throws TrackSearchException {
```
Comment (s-frei) [inline library/src/main/java/io/sfrei/tracksearch/clients/MultiSearchClient.java:47]: This shouldn't appear here...

**Review by s-frei — CHANGES_REQUESTED** [review]
Good first shot but seems to be an incomplete implementation. Either add the *SoundCloud* related functionality or make this just exclusive for the `YouTubeClient`.

Please add tests for this!

**Comment by s-frei** [comment]
Related feature request: #91

**Comment by s-frei** [comment]
It would also be pretty convenient to get the related track from another track 😉 
Will have a look at this later...

File: library/src/main/java/io/sfrei/tracksearch/tracks/deserializer/YouTubeTrackDeserializer.java (hunk 3 above)
Comment (MrF1yn) [reply library/src/main/java/io/sfrei/tracksearch/tracks/deserializer/YouTubeTrackDeserializer.java:31]: Didn't understand this.

**Comment by MrF1yn** [comment]
Alr thanks for the feedbacks.

**Comment by s-frei** [comment]
As there is no update on this... Maybe picking this anytime later.

**Pull request closed without merging**